### ISSUE TITLE
Add signal for invalid login attempt

### ISSUE THREAD

**Issue by jameshanlon** [lingthio/Flask-User#214]
This is a feature request. Flask-User is entirely vulnerable to brute force password attacks. However, it would be straight forward to add a signal for a failed login attempt to allow users of Flask-Login to implement a mechanism to defend against this.

**Comment by asselinpaul** [comment]
I would like to see this merged.

Tracking failed logins can be helpful for threat monitoring.

**Comment by jameshanlon** [comment]
@lingthio - This issue has been open since May 2018 and represents a significant security consideration for the project. A simple fix has been made and merge request open since August 2018 by @drryanjames. Is there any reason why the fix can't be merged now?

**Comment by emin-ec** [comment]
I'm curious what the status of this issue and related PR is. This seems like a good fix. Is there a reason why it seems to have been ignored? I guess an alternative is to override the login form in flask security or other tools and react to failed logins that way. It would be nice to have an official solution though or know what the issues are in not having one.

Thanks.

**Comment by and-semakin** [comment]
I agree that this is a very important functionality. The only reason it is not yet merged is the lack of maintainers time. I plan to review and merge the PRs that have accumulated over a long time, including #230. We also need write tests for this feature. Thank you for your patience :)
